test/wallet.test.ts

```diff
@@ -103,6 +103,29 @@ describe('wallet', () => {
         );
       }
     });
+
+    it('throws unauthorized for unknown addresses', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice(0, 2);
+      const witnessedTxParams: TransactionParams[] = [];
+      const processTransaction = async (_txParams: TransactionParams) => {
+        witnessedTxParams.push(_txParams);
+        return testTxHash;
+      };
+      engine.push(createWalletMiddleware({ getAccounts, processTransaction }));
+      const txParams = {
+        from: testUnkownAddress,
+      };
+
+      const payload = { method: 'eth_sendTransaction', params: [txParams] };
+      try {
+        await pify(engine.handle).call(engine, payload);
+      } catch (e: any) {
+        expect(e.message).toEqual(
+          'The requested account and/or method has not been authorized by the user.',
+        );
+      }
```

Comment:
Similar to another comment, it's better to write this as the following to rule out a false positive:

``` suggestion
      const promise = pify(engine.handle).call(engine, payload);
      await expect(promise).rejects.toThrow('The requested account and/or method has not been authorized by the user.')
```